Mathlib/Order/KrullDimension.lean

```diff
@@ -0,0 +1,92 @@
+/-
+Copyright (c) 2023 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang, Fangmin Li
+-/
+
+import Mathlib.Order.Monotone.Basic
+import Mathlib.Order.CompleteLattice
+import Mathlib.Order.WithBot
+import Mathlib.Data.Fin.Basic
+import Mathlib.Data.Nat.Lattice
+import Mathlib.Logic.Equiv.Fin
+
+/-!
+# Krull dimension of a preordered set
+
+If `α` is a preordered set, then `krull_dim α` is defined to be `sup {n | a₀ < a₁ < ... < aₙ}`.
+For `a : α`, its height is defined to be the krull dimension of the subset `(-∞, a]` while its
+coheight is defined to be the krull dimension of `[a, +∞)`.
+-/
+
+variable (α : Type _) [Preorder α]
+
+/--
+For a preordered set `(α, <)`, a strict series of `α` of length `n` is a strictly monotonic function
+`fin (n + 1) → α`, i.e. `a₀ < a₁ < ... < aₙ` with `aᵢ : α`.
+-/
+structure StrictSeries where
+/-- length of a strict series -/
+length : ℕ
```

Comment:
This seems like a pretty weird meaning of the word "length", since `toFun` has `length + 1` items.

Reply:
do you have any suggestions?

Reply:
maybe `maxIndex` is a better name

Reply:
"Length" is standard in commutative algebra for this concept -- see for example https://en.wikipedia.org/wiki/Krull_dimension#Explanation